### DISCUSSION TITLE
Possible to restrict (allow-list) what marked will parse?

### DISCUSSION THREAD

**Discussion by Saeven** [markedjs/marked#3293] [Q&A]
For example, imagine a situation where I only want to allow "bold".  What's the best approach to limiting marked's parsing to "bold" only?

Thanks!

**Comment by UziTech** [comment]
There are a couple ways depending on your use case: 
1. Create a [tokenizer](https://marked.js.org/using_pro#tokenizer) that only tokenizes **emStrong** and text tokens.
2. Create a [renderer](https://marked.js.org/using_pro#renderer) that renders every token as text except **strong**
3. Change all token types to `text` tokens in [walkTokens](https://marked.js.org/using_pro#walk-tokens) except **strong**

Depending on your use case one of those will be easier but be prepared to test lots of edge cases.

**Comment by UziTech** [comment]
If all you need are inline tokens you can use [`parseInline`](https://marked.js.org/using_advanced#inline)

**Comment by ivanjaros** [comment]
Depends on whether you want to sanitize on input or output. Personally, I run the output through dompurify library.
